### DISCUSSION TITLE
Shared embeddings layer between `ner` & `entity_linker`

### DISCUSSION THREAD

**Discussion by Joozty** [explosion/spaCy#9291] [New Features & Project Ideas]
Today I wanted to use shared [embedding layer feature](https://spacy.io/usage/embeddings-transformers#embedding-layers) in my training.  I am training `ner` + `entity_linker` pipelines. However, my training is failing on `Mismatched IDs received by the Tok2Vec listener: {id1} vs. {id2}` error.

 I looked at the code and found out that EL pipeline sends slightly different [batch](https://github.com/explosion/spaCy/blob/master/spacy/pipeline/entity_linker.py#L254) to the `tok2vec` model. Basically when there is a sentence containing two annotations it sends downs  two docs (`['sentence_entity_1', 'sentence_entity2']`) instead of one. 

So I was wondering if you plan to support this feature?

**Comment by svlandeg** [comment]
Hi! Can you share the config & training command? That will help us understand what exactly you're trying to do, replicate the issue, and see whether it's something we could address on our end or not. Thanks!

**Comment by Joozty** [comment]
Yes sure, here is an example https://github.com/Joozty/ner_nel_shared_embed_layer

**Comment by svlandeg** [comment]
This could be related to https://github.com/explosion/spaCy/discussions/9310 and it's likely a bug. I'll look into it!

**Reply by Joozty** [reply]
any luck? 🙂 Maybe this should be converted to issue?

**Comment by polm** [comment]
Thanks for your sample project and details about this - your analysis of the batches being an issue is correct, and this is a bug and the Entity Linker doesn't work with listeners. 

The listener requires components to not make certain kinds of modifications to incoming batches, but the Entity Linker predates the listener, so it didn't follow that requirement. We'll need to rework the NEL component some to make this work. 

I should be able to start on a PR to fix this soon, I'll link here when I do.

**Reply by polm** [reply]
PR is up: https://github.com/explosion/spaCy/pull/9669

There are some details of this that I still haven't figured out, but updates will go in the PR as they're ready.
